### PR TITLE
drop --max-vms, generate vm config upon startup

### LINKED ISSUE #314: drakrun: change the way of vm-X.cfg generation, drop --max-vms
We need to give more control over drakrun replication, it's too magic right now.

Proposal:

* remove `--max-vms`
* generate `vm-X.cfg` upon startup of `drakrun@X` service, not during `postinstall`
* add comment in `vm-X.cfg` to discourage from editing the file directly (template should be used instead)
* don't enable drakrun service by default
* don't start drakrun by default
* `drak-healthcheck` - show statuses of enabled `drakrun@X` services

### PR DESCRIPTION
**Pull request by icedevml**
resolve #314

### CODE REVIEW

File: drakcore/drakcore/bin/drak-healthcheck [hunk 1]
```diff
@@ -13,12 +15,10 @@ def service_runnning(info):
...
 
-def get_install_info():
-    try:
-        with open("/etc/drakrun/install.json") as conf:
-            return json.load(conf)
-    except OSError:
-        return None
+def get_enabled_drakruns():
+    for fn in os.listdir("/etc/systemd/system/default.target.wants"):
+        if re.match('^drakrun@([0-9]+)\.service$', fn):
```
Comment (chivay) [inline drakcore/drakcore/bin/drak-healthcheck:20]:
```suggestion
        if re.fullmatch('drakrun@[0-9]+\.service', fn):
```

- drop `^` `$` and use fullmatch
- remove unnecessary capture group

File: drakrun/drakrun/config.py [hunk 2]
```diff
@@ -14,7 +14,6 @@ class InstallInfo(DataClassJsonMixin):
     storage_backend: str
     disk_size: str
     iso_path: str
-    max_vms: int
```
Comment (chivay) [inline drakrun/drakrun/config.py:17]: I haven't checked, but are we 100% sure that this won't crash if we try to deserialize an old `install.json` with `max_vms ` field?

File: drakrun/drakrun/vmconf.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import os
+import re
+import logging
+
+from drakrun.storage import get_storage_backend
+from drakrun.config import ETC_DIR, LIB_DIR, InstallInfo
+
+
+def generate_vm_conf(install_info: InstallInfo, vm_id: int):
+    with open(os.path.join(ETC_DIR, 'scripts/cfg.template'), 'r') as f:
```
Comment (chivay) [inline drakrun/drakrun/vmconf.py:10]:
```suggestion
    with open(os.path.join(ETC_DIR, 'scripts', 'cfg.template'), 'r') as f:
```

File: drakrun/drakrun/vmconf.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+    storage_backend = get_storage_backend(install_info)
+
+    disks = []
+    disks.append(storage_backend.get_vm_disk_path(vm_id))
+
+    disks.append('file:{iso},hdc:cdrom,r'.format(iso=os.path.abspath(install_info.iso_path)))
+
+    if install_info.enable_unattended:
+        disks.append('file:{main_dir}/volumes/unattended.iso,hdd:cdrom,r'.format(main_dir=LIB_DIR))
```
Comment (chivay) [inline drakrun/drakrun/vmconf.py:21]: this should use path join

File: drakrun/drakrun/vmconf.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    disks = ', '.join(['"{}"'.format(disk) for disk in disks])
+
+    template = template.replace('{{ VM_ID }}', str(vm_id))
+    template = template.replace('{{ DISKS }}', disks)
+    template = template.replace('{{ VNC_PORT }}', str(6400 + vm_id))
+
+    if vm_id == 0:
+        template = re.sub('on_reboot[ ]*=(.*)', 'on_reboot = "restart"', template)
+
+    with open(os.path.join(ETC_DIR, 'configs/vm-{}.cfg'.format(vm_id)), 'w') as f:
```
Comment (chivay) [inline drakrun/drakrun/vmconf.py:32]:
```suggestion
    with open(os.path.join(ETC_DIR, 'configs', f'vm-{vm_id}.cfg'), 'w') as f:
```

**Review by chivay — CHANGES_REQUESTED** [review]
Generally LGTM, however forcing the user to enable services manually seems tedious.
I'd like to see some wrapper script that would automate enabling and bringing up `drakrun` services. eg:  `draksetup scale n`

Additionally, managing number of VMs should be addressed in the docs.

**Comment by icedevml** [comment]
@chivay done

File: drakrun/drakrun/draksetup.py [hunk 6]
```diff
@@ -342,69 +334,79 @@ def postinstall(report, generate_usermode):
...
+def postupgrade():
     with open(os.path.join(ETC_DIR, 'scripts/cfg.template'), 'r') as f:
         template = f.read()
 
-    storage_backend = get_storage_backend(install_info)
+    passwd_characters = string.ascii_letters + string.digits
+    passwd = ''.join(random.choice(passwd_characters) for i in range(20))
+    template = template.replace('{{ VNC_PASS }}', passwd)
+
+    with open(os.path.join(ETC_DIR, 'scripts/cfg.template'), 'w') as f:
```
Comment (chivay) [inline drakrun/drakrun/draksetup.py:355]:
```suggestion
    with open(os.path.join(ETC_DIR, 'scripts', 'cfg.template'), 'w') as f:
```

File: drakrun/drakrun/draksetup.py [hunk 7]
```diff
@@ -342,69 +334,79 @@ def postinstall(report, generate_usermode):
...
 
-    with open(os.path.join(ETC_DIR, 'scripts/cfg.template'), 'r') as f:
-        template = f.read()
+@click.command(help='Scale drakrun services',
+               no_args_is_help=True)
+@click.argument('scale_count',
+                type=int)
+def scale(scale_count):
+    '''Enable or disable additional parallel instances of drakrun service..
+    '''
```
Comment (chivay) [inline drakrun/drakrun/draksetup.py:396]:
```suggestion
    """Enable or disable additional parallel instances of drakrun service.."""
```

Comment (chivay) [reply drakrun/drakrun/draksetup.py:396]: Using `"""` adheres to docstring PEP

**Review by chivay — APPROVED** [review]
Still missing docs :confused: , but apart from that LGTM

**Pull request merged**